docs/reference/api-gateway/http.md

```diff
@@ -0,0 +1,174 @@
+# API Gateway
+
+#### In this document
+
+- [No Authentication](#none-auth)
+    - [Create](#create-none)
+    - [Invoke](#invoke-none)
+- [Basic Authentication](#basic-auth)
+    - [Create](#create-basic)
+    - [Invoke](#invoke-basic)
+- [Canary Function](#canary-function)
+
+<a id="none-auth"></a>
+## No Authentication
+
+<a id="create-none"></a>
+### Create API Gateways
+
+You can create an api gateway with basic authentication by sending a POST request to the following endpoint:
+
+```
+<nuclio-host-name>/api/api_gateways
+```
+With the following request body:
+```json
+{
+    "spec": {
+        "name": "<apigateway-name>",
+        "description": "some description",
+        "path": "/some/path", 
+        "authenticationMode": "none",
+        "upstreams": [
+            {
+                "kind": "nucliofunction",
+                "nucliofunction": {
+                    "name": "function-name-to-invoke"
+                },
+                "percentage": 0
+            }
+        ],
+        "host": "<apigateway-name>-<project-name>.<nuclio-host-name>"
+    },
+    "metadata": {
+        "labels": {
+            "nuclio.io/project-name": "default"
+        },
+        "name": "<apigateway-name>"
+    }
+}
+```
+
+<a id="invoke-none"></a>
+### Invoke API Gateways
+
+To invoke it, simply send a request to the created API Gateway ingress (e.g.: `<apigateway-name>-<project-name>.<nuclio-host-name>`,
+specified on the request body `spec.host` )
+
+<a id="basic-auth"></a>
+## Basic Authentication
+
+You may protect your function by setting the API Gateway with [basic authentication](https://en.wikipedia.org/wiki/Basic_access_authentication). 
```

Comment:
You can protect your function by applying [basic authentication](https://en.wikipedia.org/wiki/Basic_access_authentication) to the API Gateway.